src/defaultSelector.py

```diff
@@ -1,43 +1,56 @@
-from PySide2.QtCore import QObject
-from PySide2.QtGui import QVector3D
-from geometry import Geometry
+from rayTracing import dmnsn_ray, dmnsn_optimized_ray, dmnsn_aabb
 from signals import Signals
 from selinfo import SelectionInfo
 import openmesh as om
 import numpy as np
-import math
 from selection import Selector
+import time
 
 
 class DefaultSelector(Selector):
     def __init__(self):
         super().__init__()
 
     def select(self, los, geometry):
+        tSS = time.perf_counter()
+        # self.selectOld(los,geometry)
+        self.selectList(los, geometry)
+        # self.selectNP(los, geometry)
+
+        dtS = time.perf_counter() - tSS
+        print("Selection time, s:", dtS)
+
+    def selectList(self, los, geometry):
         if not len(geometry):
             return
-        P0Q = los[0]
-        KQ = los[1]
-        # transform to np arrays
-        K = np.array([KQ.x(), KQ.y(), KQ.z()])
-        P0 = np.array([P0Q.x(), P0Q.y(), P0Q.z()])
-
-        # geometry je lista geometrije iz koje treba izracunati selekciju
         sis = []
+        intrsectLeafs = []
         for geo in geometry:
             isInBox = True
             # 1. test bounding box
-            # 2. test mesh in geo
+            t = 99999999
+            ray = dmnsn_ray(los)
+            opt_ray = dmnsn_optimized_ray(ray)
+            intrsectLeafs.clear()
+            isInBox = geo.subdivboxtree.getIntersectedLeafs(opt_ray, t, intrsectLeafs)
```

Comment:
this applies to lines from 31-35:
This looks bit tedious:
- first I have to initialize t with some aribtrary value (how is it choosen?)
- than manually execute following:
  - create `ray`
  - create `opt_ray`
  - clear `intrsectLeafs`

would you pls consider changing api so it will be possible to use it like tihs:
```
isInBox, intrsectLeafs = geo.subdivboxtree.getIntersectedLeafs(los)
```

note "oneliner" instead of 5 lines of code. It's not a problem if it's used on one place, but if it is used on many places.....

Reply:
ray and opt_ray is replaced by a new Ray class. Parameter t was removed. The intrsectLeafs list has to be created outside of getIntersectedLeafs due to the recursion of getIntersectedLeafs.